### ISSUE TITLE
[FEATURE] Inline Keyboard menu

### ISSUE THREAD

**Issue by pterodattero** [python-telegram-bot/python-telegram-bot#2387]
Many bots (e.g. BotFather) make use of Inline Keyboard menus. When I have to make them I usually use one Conversation Handler with a status for each internal node of my menu tree, triggered by an inline query callback and a callback for each leaf node of the menu. By the way this can be kinda painful, with a lot of repetitive code, sometime hard to modify and very prone to bugs.

What about a method that allows to create this complex handler just by passing a tree-like object?
Features I think this should include are:
- automatic creation of status constants of CallbackQueryHandlers in order to build a tree conversation
- possibility of a "back" button, that moves brings back to the previous status
- possibility of changing the message text to which the inline keyboard is attached (a title of the menu level)
- paginate option for menu nodes with many children

**Comment by Bibo-Joshi** [comment]
Hi.

Interactive menus are indeed a common use case, but also need a high level of customization to the specific scenario. Indeed there have been attempts to add something like this: See #639 and #1702.

However, I do feel like something like this is out of scope for a general purpose library, because there will always be some edge-case feature that's not included forcing users to again do they're own thing and because it will take a lot of work to maintan (in #1702 I tried to make the setup highly customizable and it got messy, both in terms of implementation and of usage & that's why I ultimately closed it). Of course this is my personal opinion and the other developers might think differently about this.

We do have the [ptbcontrib library](https://github.com/python-telegram-bot/ptbcontrib) by now, though, and if you do build something that allows to easily set up inline menus, you're very welcome to PR over there! Let me also point out that PTB is/will be in a better position to build a contrib-extension for this as

* We have persistence since v12 - which was one of the reasons #639 was closed
* #1844 is on the ToDo-list which has great potential to be used in such an add-on IMO.

Also I'd like to point out that there is a [standalone package](https://pypi.org/project/python-telegram-bot-pagination/) for pagination with telegram bots that could be of use (ever tried it, though).
